### PR TITLE
[Retiarii] Rename APIs and refine documentation

### PR DESCRIPTION
**Pull request by ultmaster**

### CODE REVIEW

File: docs/en_US/NAS/retiarii/Tutorial.rst [hunk 1]
```diff
@@ -129,38 +129,37 @@ Use placehoder to make mutation easier: ``nn.Placeholder``. If you want to mutat
...
 
 Explore the Defined Model Space
 -------------------------------
 
-After model space is defined, it is time to explore this model space. Users can choose proper search and training approach to explore the model space.
+After model space is defined, it is time to explore this model space. Users can choose proper search and model evaluator to explore the model space.
 
-Create a Trainer and Exploration Strategy
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Create a Evaluator and Exploration Strategy
```
Comment (QuanluZhang) [inline docs/en_US/NAS/retiarii/Tutorial.rst:147]: a -> an

Comment (QuanluZhang) [reply docs/en_US/NAS/retiarii/Tutorial.rst:147]: i will revise the doc in another pr

File: nni/retiarii/evaluator/pytorch/__init__.py [hunk 2]
```diff
@@ -0,0 +1,2 @@
+from .base import PyTorchImageClassificationTrainer, PyTorchMultiModelTrainer
```
Comment (QuanluZhang) [inline nni/retiarii/evaluator/pytorch/__init__.py:1]: why not change `Trainer` to `Evaluator`?

File: nni/retiarii/evaluator/pytorch/lightning.py [hunk 3]
```diff
@@ -22,11 +22,11 @@ def set_model(self, model):
...
 
 
-Trainer = blackbox_module(pl.Trainer)
-DataLoader = blackbox_module(DataLoader)
+Trainer = serialize_cls(pl.Trainer)
+DataLoader = serialize_cls(DataLoader)
 
 
-class Lightning(TrainingConfig):
+class Lightning(ModelEvaluator):
```
Comment (QuanluZhang) [inline nni/retiarii/evaluator/pytorch/lightning.py:29]: what is the difference between `ModelEvaluator`, `Evaluator` and `Trainer`?

File: nni/retiarii/execution/logical_optimizer/logical_plan.py [hunk 4]
```diff
@@ -181,25 +181,25 @@ def assemble(self, multi_model_placement: Dict[Model, PhysicalDevice]) \
...
                     # "use_input" and "use_output" are used to mark whether
                     # an input/output of a model is used in a multi-model
                     if new_node.operation.type == '_inputs':
                         input_slot_mapping[new_node] = slot
-                        phy_model.training_config.kwargs['model_kwargs'][slot]['use_input'] = True
+                        phy_model.evaluator.kwargs['model_kwargs'][slot]['use_input'] = True
                     if new_node.operation.type == '_outputs':
                         output_slot_mapping[new_node] = slot
-                        phy_model.training_config.kwargs['model_kwargs'][slot]['use_output'] = True
+                        phy_model.evaluator.kwargs['model_kwargs'][slot]['use_output'] = True
```
Comment (QuanluZhang) [inline nni/retiarii/execution/logical_optimizer/logical_plan.py:202]: @hzhua could you please help review this part?

**Review by QuanluZhang — APPROVED** [review]

File: nni/retiarii/execution/logical_optimizer/logical_plan.py (hunk 4 above)
Comment (hzhua) [reply nni/retiarii/execution/logical_optimizer/logical_plan.py:202]: No problem if evaluator has the same structure with training_config.

File: nni/retiarii/evaluator/pytorch/__init__.py (hunk 2 above)
Comment (ultmaster) [reply nni/retiarii/evaluator/pytorch/__init__.py:1]: The base file is deprecated. These are not public APIs.

File: nni/retiarii/evaluator/pytorch/lightning.py (hunk 3 above)
Comment (ultmaster) [reply nni/retiarii/evaluator/pytorch/lightning.py]: Maybe we can deprecate the concept of trainer and rename DartsTrainer -> DARTS, EnasTrainer -> ENAS? Leave that for discussion.

**Review by liuzhe-lz — APPROVED** [review]

**Pull request merged**